### PR TITLE
base: load_menus optimization

### PR DESCRIPTION
**Pull request by yelizariev**
1.
[REM] base: delete all_menu_ids from load_menus
it's not used since Odoo 11
https://github.com/odoo/odoo/commit/13f84b9aca163a8d295678f234ddc0f5b9407111

2.
[IMP] base: load_menus: exclude child_of query
less queries -> more speed

Performance measurements (odoo and browser are running on the same machine):

```
|                                | Number of queries | Query time, sec | Remaining time, sec |
|--------------------------------+-------------------+-----------------+---------------------|
| Before (single query via cURL) |                18 |           0.042 |               0.113 |
| Before (page loading)          |                18 |           0.129 |               0.408 |
| After (single query via cURL)  |                14 |           0.025 |               0.095 |
| After (page loading)           |                14 |           0.132 |               0.255 |
```

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by yelizariev** [comment]
WIP

**Comment by yelizariev** [comment]
Similar changes in website menus:

https://github.com/odoo/odoo/commit/f95af92ae3bc6bd7d7d4a9e09042bcca9f35ffec
https://github.com/odoo/odoo/commit/b98d9c968cb7147af6c9042f0f0639d41b565fd7

**Comment by yelizariev** [comment]
Recent refactoring about load_menus: https://github.com/odoo/odoo/commit/0573acae2306bf5da2005852da9323ddc59e5431

**Review by kebeclibre — APPROVED** [review]

File: odoo/addons/base/models/ir_ui_menu.py [hunk 1]
```diff
@@ -231,40 +230,32 @@ def load_menus(self, debug):
...
+            'children': [],
         }
-
         all_menus = {'root': menu_root}
 
-        if not menu_roots_data:
-            return all_menus
-
         # menus are loaded fully unlike a regular tree view, cause there are a
         # limited number of items (752 when all 6.1 addons are installed)
```
Comment (yelizariev) [inline odoo/addons/base/models/ir_ui_menu.py:242]: I'd keep this ancient comment for archaeologists

**Comment by yelizariev** [comment]
Screw it. The PR seems still useful, but benchmarks should be measured again. Since I don't work in perf anymore and because the patch haven't got much attention, it makes sense for me to close it.

@odoo/r-d-perf-team-support

**Pull request closed without merging**